### DISCUSSION TITLE
Select doesn't work for relations with globalScope

### DISCUSSION THREAD

**Discussion by GertjanRoke** [laravel/framework#40201] [Q&A]
- Laravel Version: 8.77.1
- PHP Version: 8.0.9|8.1.0
- Database Driver & Version: MySQL 8.0.26

### Description:
I'm trying to eagerload a relationship with a custom `->select()` but this doesn't clear the earlier set selects.

### Steps To Reproduce:

I created a repo that can be spin up with Sail and has the code to reproduce this issue, in the [readme](https://github.com/GertjanRoke/scope-issue#readme) I described the issue and what result I would expect.

Repo: https://github.com/GertjanRoke/scope-issue

----

I would also love to make a PR, but I have no idea where to start or which test to look at, so I can create a test that will fail to point out the issue.

PS: I have no idea if this is a problem with the Mysql driver or with all.

**Comment by GertjanRoke** [comment]
@driesvints did you also read the original issue I created? Because I really believe this is a bug inside the building of the eager loading queries.

I don't know what the new mindset is behind putting it to Discussions, can I read that somewhere, or could you help me understand it?

**Reply by driesvints** [reply]
I didn't think this was a bug report and suggest you try a few support channels first if you don't get a reply through GitHub Discussions:

- [Laracasts Forums](https://laracasts.com/discuss)
- [Laravel.io Forums](https://laravel.io/forum)
- [StackOverflow](https://stackoverflow.com/questions/tagged/laravel)
- [Discord](https://discordapp.com/invite/KxwQuKb)
- [Larachat](https://larachat.co)
- [IRC](https://web.libera.chat/?nick=laravelnewbie&channels=#laravel)
